### DISCUSSION TITLE
AnnotationDriver does not exist

### DISCUSSION THREAD

**Discussion by DIICA99** [symfony/symfony#46403] [Q&A]
Hello,

I'm new to Symfony and I created a 4.4.41 version Symfony project with Composer version 2.3.5.

this is the command I ran:

`composer create-project symfony/website-skeleton:"^4.4" `codape_web_app`

Everything was working fine until I try to create an entity. I was provided with an error saying :

`Class Doctrine\Persistence\Mapping\Driver\AnnotationDriver does not exist` 

So I tried to install annotations via composer (composer req annotations) and this is the output:

`C:\LAMP\Apache\htdocs\symfony\codape_web_app>composer req annotations
Info from https://repo.packagist.org: #StandWithUkraine
Using version ^6.2 for sensio/framework-extra-bundle
./composer.json has been updated
Running composer update sensio/framework-extra-bundle
Loading composer repositories with package information
Updating dependencies
Nothing to modify in lock file
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Generating autoload files
111 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install public [OK]

C:\LAMP\Apache\htdocs\symfony\codape_web_app>
`
And this is the error that I get when trying to create any entity:

`C:\LAMP\Apache\htdocs\symfony\codape_web_app>php bin/console make:entity Activite

 Your entity already exists! So let's add some new fields!

 New property name (press <return> to stop adding fields):
 > newfield

 Field type (enter ? to see all types) [string]:
 > ?

Main types
  * string
  * text
  * boolean
  * integer (or smallint, bigint)
  * float

Relationships / Associations
  * relation (a wizard will help you build the relation)
  * ManyToOne
  * OneToMany
  * ManyToMany
  * OneToOne

Array/Object Types
  * array (or simple_array)
  * json
  * object
  * binary
  * blob

Date/Time Types
  * datetime (or datetime_immutable)
  * datetimetz (or datetimetz_immutable)
  * date (or date_immutable)
  * time (or time_immutable)
  * dateinterval

Other Types
  * ascii_string
  * decimal
  * guid
  * json_array

 Field type (enter ? to see all types) [string]:
 > relation

In DoctrineHelper.php line 149:
                                                                             
  Class Doctrine\Persistence\Mapping\Driver\AnnotationDriver does not exist  
                                                                             

make:entity [-a|--api-resource] [--regenerate] [--overwrite] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--]
 <command> [<name>]
`
This is my composer.json

{
    "type": "project",
    "license": "proprietary",
    "require": {
        "php": ">=7.1.3",
        "ext-ctype": "*",
        "ext-iconv": "*",
        "doctrine/annotations": "^1.0",
        "doctrine/dbal": "2.13.1",
        "doctrine/doctrine-bundle": "^2.6",
        "doctrine/doctrine-migrations-bundle": "^3.2",
        "doctrine/orm": "^2.12",
        "phpdocumentor/reflection-docblock": "^5.3",
        "sensio/framework-extra-bundle": "^6.2",
        "symfony/apache-pack": "^1.0",
        "symfony/asset": "4.4.*",
        "symfony/console": "4.4.*",
        "symfony/dotenv": "4.4.*",
        "symfony/expression-language": "4.4.*",
        "symfony/flex": "^1.3.1",
        "symfony/form": "4.4.*",
        "symfony/framework-bundle": "4.4.*",
        "symfony/http-client": "4.4.*",
        "symfony/intl": "4.4.*",
        "symfony/mailer": "4.4.*",
        "symfony/monolog-bundle": "^3.1",
        "symfony/process": "4.4.*",
        "symfony/property-access": "4.4.*",
        "symfony/property-info": "4.4.*",
        "symfony/proxy-manager-bridge": "4.4.*",
        "symfony/security-bundle": "4.4.*",
        "symfony/serializer": "4.4.*",
        "symfony/translation": "4.4.*",
        "symfony/twig-bundle": "4.4.*",
        "symfony/validator": "4.4.*",
        "symfony/web-link": "4.4.*",
        "symfony/webpack-encore-bundle": "^1.14",
        "symfony/yaml": "4.4.*",
        "twig/extra-bundle": "^2.12|^3.0",
        "twig/twig": "^2.12|^3.0"
    },
    "require-dev": {
        "phpunit/phpunit": "^9.5",
        "symfony/browser-kit": "4.4.*",
        "symfony/css-selector": "4.4.*",
        "symfony/debug-bundle": "4.4.*",
        "symfony/maker-bundle": "1.30.2",
        "symfony/phpunit-bridge": "^6.0",
        "symfony/stopwatch": "4.4.*",
        "symfony/web-profiler-bundle": "4.4.*"
    },
    "config": {
        "allow-plugins": {
            "composer/package-versions-deprecated": true,
            "symfony/flex": true
        },
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "replace": {
        "paragonie/random_compat": "2.*",
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-php71": "*",
        "symfony/polyfill-php70": "*",
        "symfony/polyfill-php56": "*"
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "allow-contrib": true,
            "require": "*"
        }
    }
}

Now what I want to know is why the `Class Doctrine\Persistence\Mapping\Driver\AnnotationDriver does not exist` occures and how to solve it.

I really need help here...
Sorry for the formating, this is my first GitHub discussion

**Comment by DylanKas** [comment]


**Reply by DIICA99** [reply]
I have the 2.10 version

`"doctrine/orm": "^2.10 || ^3.0",`

**Comment by DylanKas** [comment]
So I checked online for a similar issue and [I found this](https://github.com/symfony/maker-bundle/issues/1090).

You need to change your composer.json and downgrade your doctrine/orm to 2.11 until you migrate to symfony 5.4 because [symfony 4.4 support has been dropped](https://github.com/symfony/maker-bundle/pull/1098).

So modify your composer.json with:

`"doctrine/orm": "^2.11",`

And also:

```
"conflict": {
        "symfony/symfony": "*",
        "doctrine/orm": "2.12.0"
    },
```

Don't forget to run `composer update`
